### PR TITLE
[TheiaMeta] Remove subworkflow for assembly with metaspades and pilon; remove exposed krakendb input from QC subworkflow

### LINKED ISSUE #337: [TheiaMeta] Kraken2 inputs duplicated due to PR #286
:bug:

### :pencil: Describe the Issue

Since the introduction of #286 the kraken2 inputs are confusing on the TheiaMeta workflow
For example, there's two places that accept a kraken2 database:

image
image

Only the first one controls the kraken2 calls for the raw and clean reads.

image

These inputs should either be adopted or suppressed in the TheiaMeta workflow logic.

### LINKED ISSUE #364: [TheiaMeta] cpu, docker, memory and disk_size not exposed to Terra for metaspades
:bug:

### :repeat: How to Reproduce

- Was this run on the Terra platform?
  - Terra on GCP or Azure?
- Or were the workflows run locally using `miniwdl` or `cromwell`?
  - **What was the exact command was used to launch the workflow?**
- If possible, we may ask you to share your Terra workspace with us. Usually READER access is sufficient, but we may ask for WRITER access if we need to make changes to the workspace to reproduce the issue.

### PR DESCRIPTION
**Pull request by cimendes**
This PR closes #364 and closes #337.

🗑️ This dev branch should <NOT> be deleted after merging to main.

## :brain: Aim, Context and Functionality

This PR removes the metaspades assembly subworkflow in favour of incorporating it within the theiameta workflow. This reduces the number of dependencies needed and makes the inputs a little bit more neat. 

Additionally, PR #279 added kraken as part of the QC trim sub-workflow. This created conflicts and confusion within TheiaMeta that already incorporates kraken on both raw and clean-reads. The inputs have been hidden to facilitate usage of TheiaMeta. 

## :hammer_and_wrench:  Impacted Workflows/Tasks & Changes Being Made
This will affect the behavior of the workflow(s) even if users don’t change any workflow inputs relative to the last version : No

Running this workflow on different occasions could result in different results, e.g. due to use of a live database, "latest" docker image, or stochastic data processing : No 

#### 🔄 Data Processing 

Nothing has been changed.

Docker/software or software versions changed:  None

Databases or database versions changed: None

Data processing/commands changed: None

File processing changed: None

Compute resources changed: None

#### ➡️ Inputs 

Nothing has been changed.

#### ⬅️ Outputs 

Nothing has been changed.

#### Commandline Testing with MiniWDL or Cromwell (optional)

Not performed! 

#### Terra Testing

Inputs show up as expected!
image

-[ TheiaMeta Illumina PE Functional Test](https://app.terra.bio/#workspaces/theiagen-validations/Theiagen_Mendes_Sandbox/job_history/a8a9d488-4560-41fa-b7c2-0f1012792500)

## :microscope: Final Developer Checklist

- [ ] The workflow/task has been tested locally and results, including file contents, are as anticipated
- [x] The workflow/task has been tested on Terra and results, including file contents, are as anticipated
- [x] The CI/CD has been adjusted and tests are passing (to be completed by Theiagen developer)
- [x] Code changes follow the [style guide](https://theiagen.notion.site/Style-Guide-WDL-Workflow-Development-bb456f34322d4f4db699d4029050481c)

## 🎯 Reviewer Checklist 

- [x] All impacted workflows/tasks have been tested on Terra with a different dataset than used for development
- [x] All reviewer-suggested scenarios have been tested and any additional
- [x] All changed results have been confirmed to be accurate
- [x] All workflows/tasks impacted by change/s have been tested using a standard validation dataset to ensure no unintended change of functionality
- [x] All code adheres to the style guide
- [x] MD5 sums have been updated
- [x] The PR author has addressed all comments

## 🗂️ Associated Documentation (to be completed by Theiagen developer)

- [ ] Relevant documentation on the Public Health Resources "PHB Main" has been updated
- [ ] Workflow diagrams have been updated to reflect changes

### CODE REVIEW

File: workflows/theiameta/wf_theiameta_illumina_pe.wdl [hunk 1]
```diff
@@ -60,18 +62,39 @@ workflow theiameta_illumina_pe {
       kraken2_report = kraken2_clean.kraken2_report,
       samplename = samplename
   }
-  call metaspades_assembly_wf.metaspades_assembly_pe as metaspades {
+  call metaspades_task.metaspades_pe {
     input:
-      read1 = read_QC_trim.read1_clean,
-      read2 = read_QC_trim.read2_clean,
+      read1_cleaned = read1,
```
Comment (kevinlibuit) [inline workflows/theiameta/wf_theiameta_illumina_pe.wdl:67]: Should these task inputs be set to read_QC_trim outputs? I.e. 
```
  call metaspades_task.metaspades_pe {
    input:
      read1 = read_QC_trim.read1_clean,
      read2 = read_QC_trim.read2_clean,
      samplename = samplename
    }
```

File: workflows/theiameta/wf_theiameta_illumina_pe.wdl [hunk 2]
```diff
@@ -60,18 +62,39 @@ workflow theiameta_illumina_pe {
...
+      read2_cleaned = read2,
       samplename = samplename
-    }
+  }
+  call minimap2_task.minimap2 as minimap2_assembly_correction {
+    input:
+      query1 = read1,
```
Comment (kevinlibuit) [inline workflows/theiameta/wf_theiameta_illumina_pe.wdl:73]: Same comment as above RE read_QC_trim outputs

**Comment by cimendes** [comment]
> With these changes, is there a need to maintain the `wf_metaspades_assembly.wdl` subworkflow?

None at all! Will remove :)

File: workflows/theiameta/wf_theiameta_illumina_pe.wdl (hunk 1 above)
Comment (cimendes) [reply workflows/theiameta/wf_theiameta_illumina_pe.wdl]: Oh my god! Thank you!

**Review by kevinlibuit — COMMENTED** [review]
Changes look great and I'm seeing the krakendb input on the Terra UI as expected. Just launched a [function test](https://app.terra.bio/#workspaces/theiagen-validations/Libui-Sandbox_2023/job_history/e45468c8-2d5c-451d-91dc-a770a0823456) on my end and will approve pending a successful run.

**Review by kevinlibuit — APPROVED** [review]

**Comment by kevinlibuit** [comment]
Ready for merge :)

**Comment by kevinlibuit** [comment]
Now for docs!

**Pull request merged**